src/test/java/com/atlassian/db/replica/internal/AuroraPostgresLsnReplicaConsistencyTest.java

```diff
@@ -0,0 +1,117 @@
+import com.atlassian.db.replica.internal.util.ConnectionSupplier;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public class AuroraPostgresLsnReplicaConsistencyTest {
+
+    private static final long LAST_WRITE_LSN = 8679792506L;
+
+    private VolatileCache<Long> lastWriteCache;
+    private AuroraPostgresLsnReplicaConsistency consistency;
+    private Connection main;
+    private Connection replica;
+
+    @Before
+    public void setUp() {
+        lastWriteCache = new VolatileCache<>();
+        consistency = new AuroraPostgresLsnReplicaConsistency.Builder()
+            .cacheLastWrite(lastWriteCache)
+            .build();
+
+        main = mock(Connection.class);
+        replica = mock(Connection.class);
+    }
+
+    @Test
+    public void shouldBeInconsistentWhenLastWriteLsnUnknown() {
+        lastWriteCache.reset();
+
+        boolean isConsistent = consistency.isConsistent(new ConnectionSupplier(replica));
+
+        assertThat(isConsistent).isFalse();
+    }
+
+    @Test
+    public void shouldBeInconsistentWhenLastWriteLsnFails() throws Exception {
+        lastWriteCache.put(LAST_WRITE_LSN);
+
+        mockLsnFetchingFailure(main);
+        mockLsnFetching(replica, LAST_WRITE_LSN + 1);
+        consistency.write(main);
+
+        boolean isConsistent = consistency.isConsistent(new ConnectionSupplier(replica));
+
+        assertThat(isConsistent).isFalse();
+    }
+
+    @Test
+    public void shouldBeInconsistentWhenReplicaLsnFails() throws Exception {
+        mockLsnFetching(main, LAST_WRITE_LSN);
+        mockLsnFetchingFailure(replica);
+        consistency.write(main);
+
+        boolean isConsistent = consistency.isConsistent(new ConnectionSupplier(replica));
+
+        assertThat(isConsistent).isFalse();
+    }
+
+    @Test
+    public void shouldBeInconsistentWhenReplicaLsnLowerThanLastWriteLsn() throws Exception {
+        mockLsnFetching(main, LAST_WRITE_LSN);
+        mockLsnFetching(replica, LAST_WRITE_LSN - 1);
+        consistency.write(main);
+
+        boolean isConsistent = consistency.isConsistent(new ConnectionSupplier(replica));
+
+        assertThat(isConsistent).isFalse();
+    }
+
+    @Test
+    public void shouldBeConsistentWhenReplicaLsnGreaterThanLastWriteLsn() throws Exception {
+        mockLsnFetching(main, LAST_WRITE_LSN);
+        mockLsnFetching(replica, LAST_WRITE_LSN + 1);
+        consistency.write(main);
+
+        boolean isConsistent = consistency.isConsistent(new ConnectionSupplier(replica));
+
+        assertThat(isConsistent).isTrue();
+    }
+
+    @Test
+    public void shouldBeConsistentWhenReplicaLsnEqualToLastWriteLsn() throws Exception {
```

Comment:
Tests can use more business-level language instead of the technical implementation details:
```suggestion
    public void shouldBeConsistentWhenReplicaCaughtUp() throws Exception {
```

Accompanying test cases:
`shouldBeConsistentWhenReplicaIsAhead`
`shouldBeInconsistentWhenReplicaIsBehind`